File: backend/src/main/java/indiana/indi/indiana/dto/games/CardItemDto.java
package indiana.indi.indiana.dto.games;

import java.math.BigDecimal;

public record CardItemDto(
        Long id,
        String title,
        String imageUrl,
        BigDecimal price,
        Integer discountPercent,
        BigDecimal finalPrice,
        Integer rating,
        Long categoryId,
        boolean isFavorite,
        boolean isInCart,
        boolean isPurchased) {
}

File: frontend/src/types/games/CardItemDto.ts
export type CardItemDto = {
    id: number;
    title: string;
    imageUrl: string;
    price: number;
    discountPercent: number;
    finalPrice: number;
    rating: number;
    isFavorite: boolean;
    isInCart: boolean;
    isPurchased: boolean;
}
